src/main/java/ru/yandex/practicum/filmorate/service/ReviewService.java

```diff
@@ -0,0 +1,86 @@
+package ru.yandex.practicum.filmorate.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import ru.yandex.practicum.filmorate.exception.NotFoundException;
+import ru.yandex.practicum.filmorate.model.Review;
+import ru.yandex.practicum.filmorate.model.enums.EventOperation;
+import ru.yandex.practicum.filmorate.model.enums.EventType;
+import ru.yandex.practicum.filmorate.storage.feed.FeedStorage;
+import ru.yandex.practicum.filmorate.storage.film.FilmStorage;
+import ru.yandex.practicum.filmorate.storage.review.ReviewStorage;
+import ru.yandex.practicum.filmorate.storage.user.UserStorage;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ReviewService {
+
+    private final ReviewStorage reviewStorage;
+    private final UserStorage userStorage;
+    private final FilmStorage filmStorage;
+    private final FeedStorage feedStorage;
+
+    public Review addReview(Review review) {
+        validateReview(review);
+        Review addReview = reviewStorage.addReview(review);
+        log.info("Добавление отзыва с Id {}", addReview.getReviewId());
+        feedStorage.addEvent(addReview.getUserId(), addReview.getReviewId(), EventOperation.ADD, EventType.REVIEW);
+
+        return addReview;
+    }
+
+    public Review updateReview(Review review) {
+        log.info("Обновление отзыва с Id {}", review.getReviewId());
+        validateReview(review);
+        Review updateReview = reviewStorage.updateReview(review)
+                .orElseThrow(() -> new NotFoundException("Отзыв с ID " + review.getReviewId() + " не найден"));
+        feedStorage.addEvent(updateReview.getUserId(), updateReview.getReviewId(), EventOperation.UPDATE,
+                EventType.REVIEW);
+        return updateReview;
+    }
+
+    public void deleteReview(Long reviewId) {
+        log.info("Удаление отзыва с reviewId {}", reviewId);
+        List<Review> review = reviewStorage.findById(reviewId).stream().collect(Collectors.toList());
```

Comment:
Стримы принято бить по операциям, чтобы улучшить читаемость:
```suggestion
        List<Review> review = reviewStorage.findById(reviewId).stream()
            .collect(Collectors.toList());
```